src/browser/api_protocol/api_handlers/application.js

```diff
@@ -383,6 +384,67 @@ function runApplication(identity, message, ack, nack) {
     }
 }
 
+async function runApplications(identity, message, ack, nack) {
+    const { payload } = message;
+    const { applications } = payload;
+
+    const appsToRunWithRVM = [];
+
+    async function runManifest(application) {
+        const { uuid, manifestUrl } = application;
+
+        const appIdentity = apiProtocolBase.getTargetApplicationIdentity(application);
+        let remoteSubscriptionUnSubscribe;
+        const logMsg = manifestUrl ?
+            `unsub called before duplicate uuid transport listener removed for ${uuid}` :
+            '';
+        /*jshint loopfunc: true */
+        let unsub = () => writeToLog('error', logMsg);
+        const remoteSubscription = {
+            uuid,
+            name: uuid,
+            listenType: 'once',
+            className: 'window',
+            eventName: 'fire-constructor-callback'
+        };
+
+        if (coreState.getAppRunningState(uuid)) {
+            Application.emitRunRequested(appIdentity);
```

Comment:
:neckbeard: reads a bit better if this is at the top (stuff above it not necessary to figure out if you need early return)

Reply:
Moved.